File: clients/elixir/puid.ex
# PUID client (Elixir). Requires :req. Auth via PUID_API_KEY.
defmodule Puid do
  @base "https://puid.dev/api"
  defp hdr, do: [{"X-API-Key", System.get_env("PUID_API_KEY") || ""}]
  def generate(n \\ 1) when n in 1..10 do
    case Req.get!("#{@base}/v1/ids?n=#{n}", headers: hdr()) do
      %{status: 429} -> {:error, :rate_limited}
      %{body: %{"ids" => ids}} -> {:ok, ids}
    end
  end
  def ordinal(puid), do: Req.get!("#{@base}/v1/ordinal/#{puid}", headers: hdr()).body["ordinal"]
end

File: clients/go/puid.go
// Package puid is the Go client for PUID. Auth via PUID_API_KEY.
package puid

import (
	"encoding/json"
	"errors"
	"fmt"
	"net/http"
	"net/url"
	"os"
)

const Base = "https://puid.dev/api"

func get(path string) (*http.Response, error) {
	req, _ := http.NewRequest("GET", Base+path, nil)
	req.Header.Set("X-API-Key", os.Getenv("PUID_API_KEY"))
	return http.DefaultClient.Do(req)
}
func Generate(n int) ([]string, error) {
	if n < 1 || n > 10 {
		return nil, errors.New("n must be 1..10")
	}
	resp, err := get(fmt.Sprintf("/v1/ids?n=%d", n))
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
	if resp.StatusCode == http.StatusTooManyRequests {
		return nil, errors.New("rate limited: one per second")
	}
	var out struct {
		IDs []string `json:"ids"`
	}
	return out.IDs, json.NewDecoder(resp.Body).Decode(&out)
}
func Ordinal(puid string) (string, error) {
	resp, err := get("/v1/ordinal/" + url.PathEscape(puid))
	if err != nil {
		return "", err
	}
	defer resp.Body.Close()
	var out struct {
		Ordinal string `json:"ordinal"`
	}
	return out.Ordinal, json.NewDecoder(resp.Body).Decode(&out)
}
